docs_src/microservices/application/ContextAPI.md

```diff
@@ -1,108 +1,142 @@
 
-The `EventClient ` exposed on the context is available to leverage Core Data's `Event` API. See [interface definition](https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/clients/coredata/event.go#L35) for more details. This client is useful for querying events and is used by the [PushToCore](#pushtocore) convenience API described below.
+### CommandClient() 
 
-### ValueDescriptorClient
+Returns a `CommandClient`  to leverage Core Command's `Command` API. See [interface definition](https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/clients/interfaces/command.go) for more details. Useful for sending commands to devices.
 
-The `ValueDescriptorClient ` exposed on the context is available to leverage Core Data's `ValueDescriptor` API. See [interface definition](https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/clients/coredata/value_descriptor.go#L29) for more details. Useful for looking up the value descriptor for a reading received.
+### NotificationClient()
 
-### CommandClient 
+Returns a `NotificationClient` to leverage Support Notifications' `Notifications` API. See [README](https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/clients/interfaces/notification.go) for more details. Useful for sending notifications. 
 
-The `CommandClient ` exposed on the context is available to leverage Core Command's `Command` API. See [interface definition](https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/clients/command/client.go#L28) for more details. Useful for sending commands to devices.
+### SubscriptionClient()
+Returns a `SubscriptionClient` to leverage Support Notifications' `Subscription` API. Note if Support Notifications is not specified in the Clients configuration, this will return nil.
 
-### NotificationsClient
+### DeviceServiceClient()
+Returns a `DeviceServiceClient` to leverage Core Metadata's `DeviceService` API. Note if Core Metadata is not specified in the Clients configuration, this will return nil.
 
-The `NotificationsClient` exposed on the context is available to leverage Support Notifications' `Notifications` API. See [README](https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/clients/notifications/README.md) for more details. Useful for sending notifications. 
+### DeviceProfileClient()
+Returns a `DeviceProfileClient` to leverage Core Metadata's `DeviceProfile` API. Note if Core Metadata is not specified in the Clients configuration, this will return nil.
+
+### DeviceClient()
+Returns a `DeviceClient` to leverage Core Metadata's `Device` API. Note if Core Metadata is not specified in the Clients configuration, this will return nil.
```

Comment:
```suggestion
Returns a `DeviceClient` to leverage Core Metadata's `Device` API. See [interface definition](https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/clients/http/device.go) for more details. Useful for querying information about Devices. Note if Core Metadata is not specified in the Clients configuration, this will return nil. 
```